git/git.go

```diff
@@ -40,6 +40,7 @@ type CloneRepoOptions struct {
 	Depth        int
 	CABundle     []byte
 	ProxyOptions transport.ProxyOptions
+	Logf         log.Func
```

Comment:
Let's change this to `func(line string)` (or printf signature if you prefer).

I don't want us logging `#1` in the clone (or package for that matter) as it should not have any context of where/when it runs.

We should probably fix other instances of similar behavior as well.

Reply:
@mafredri if I understand correctly, your idea is to use the return value of `startStage()` instead? I think that's reasonable.